feat: expose `cwd` in input options

jwnx opened the pull request:
Exposes `cwd` prop in `InputOptions` and updates rolldown tests to
leverage it. Closes #687

jwnx commented:
@hyf0 I refactored rolldown tests to use `cwd` for testing purposes, but if you find it unnecessary I can remove this change.

hyf0 commented on packages/rolldown/tests/fixtures/input/array/_config.ts:
```diff
@@ -4,7 +4,8 @@ import { expect } from 'vitest'
 const config: RollupOptions = {
-  input: [path.join(__dirname, 'main.js'), path.join(__dirname, 'entry.js')],
+  input: ['main.js', 'entry.js'],
+  cwd: __dirname,
```
Since you already pass `cwd` in `packages/rolldown/tests/fixture.test.ts`, this is unnessary to pass it again.

jwnx replied:
Done!

hyf0 commented on packages/rolldown/tests/fixture.test.ts:
You could also remove this line

https://github.com/rolldown/rolldown/blob/31926bd88f50765b9da113ee545ec7e4c0b439a1/packages/rolldown/tests/fixture.test.ts#L37

Rolldown will emit to `<CWD>/dist` by default.

jwnx replied:
Removed

jwnx committed:
feat: expose `cwd` in input options

Exposes `cwd` prop in `InputOptions` and updates rolldown tests to
leverage it. Closes #687

jwnx committed:
refactor(test): remove unnecessary path manipulation

jwnx commented on packages/rolldown/tests/cli/fixtures/basic/rolldown.config.js:
@hyf0 I was able to remove a lot of path manipulation from this file and tests are passing just fine. Let me know if you think we should keep the old code.

hyf0 reviewed (approved):
Well done!
